pkg/asset/userprovided.go

```diff
@@ -24,7 +24,8 @@ func (a *UserProvided) Dependencies() []Asset {
 }
 
 // Generate queries for input from the user.
-func (a *UserProvided) Generate(map[Asset]*State) (state *State, err error) {
+func (a *UserProvided) Generate(map[string]*State) (*State, error) {
```

Comment:
this change seems orthogonal.

Reply:
Only part of it. The map in the argument does need to change. Fixed.